### DISCUSSION TITLE
metrics stay zero

### DISCUSSION THREAD

**Discussion by henkiewie** [containrrr/watchtower#1493] [Q&A] [answered]
Hi,
i've setup the watchtower API to use metrics. When I run the example command I can see its working with the token.
The end result is;
`time="2022-12-06T09:06:29+01:00" level=info msg="Session done" Failed=0 Scanned=27 Updated=0 notify=no`
But when I ask for the metrics the number of scanned containers stays zero.
`watchtower_containers_scanned 0`
I also turned on trace to see some errors but there seem to be none.
I've set the local time volume.
What am i missing?

**Comment by piksel** [comment, marked as answer]
`watchtower_containers_scanned` is a gauge, meaning it gets reset on every session. 

Tried it just now, and it works as expected:

<details>
<summary><code>docker run --rm -it -p 8080:8080 -v /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --http-api-metrics --http-api-token t --interval 10</code></summary>

```
INFO[0001] Watchtower 1.5.1
INFO[0001] Using no notifications
INFO[0001] Checking all containers (except explicitly disabled with label)
INFO[0001] Scheduling first run: 2022-12-06 10:10:35 +0000 UTC
INFO[0001] Note that the first check will be performed in 9 seconds
INFO[0017] Session done                                  Failed=0 Scanned=1 Updated=0 notify=no
INFO[0021] Session done                                  Failed=0 Scanned=1 Updated=0 notify=no
```

</details>

<details>
<summary><code>curl -H 'Authorization: Bearer t' localhost:8080/v1/metrics | grep -v \# | grep watchtower_</code></summary>

```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  6806    0  6806    0     0  3323k      0 --:--:-- --:--:-- --:--:-- 3323k
watchtower_containers_failed 0
watchtower_containers_scanned 1
watchtower_containers_updated 0
watchtower_scans_skipped 0
watchtower_scans_total 2
```

</details>

The report (in the "Session done" log message) is built from the metrics data, so they should always be aligned. Perhaps you connected to another watchtower instance, or the result got cached somehow?

**Reply by henkiewie** [reply]
Thanks for the quick reply.
I was looking at the wrong parameters.
my result;
```watchtower_containers_failed 0
watchtower_containers_scanned 31
watchtower_containers_updated 0
watchtower_scans_skipped 0
watchtower_scans_total 1
```

**Reply by simonaronsson** [reply]
In hindsight, a gauge might have been a poor choice for these metrics. It should probably have been a monotonically increasing counter instead. We can't really change this now though as it would break things for all current users. Might be worth to consider adding the counters as `<same_name>_count` at some point though.
